java/code/src/com/redhat/rhn/manager/contentmgmt/ContentManager.java

```diff
@@ -603,7 +604,7 @@ public void promoteProject(String projectLabel, String envLabel, boolean async,
         // if current Environment or the next 2 Environments in the chain are building -> FORBID promote
         // as it could affect the build in progress
         if (isEnvironmentBuilding(of(env)) || isEnvironmentBuilding(of(nextEnv)) ||
-                isEnvironmentBuilding(nextEnv.getNextEnvironmentOpt())) {
+                BooleanUtils.isTrue(isEnvironmentBuilding(nextEnv.getNextEnvironmentOpt()))) {
```

Comment:
Not sure why SonarCloud is complaining only on the last occurrence but `isEnvironmentBuilding()` will never return null so we can just change the return type to `boolean`:

```java
    private Boolean isEnvironmentBuilding(Optional<ContentEnvironment> env) {
        return env.flatMap(e -> e.computeStatus().map(status -> status.equals(EnvironmentTarget.Status.BUILDING)))
                .orElse(false);
    }
```